test/virtual/include/openenclave/edger8r/common.h

```diff
@@ -153,6 +153,34 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
 
 #define OE_WRITE_IN_OUT_PARAM OE_WRITE_IN_PARAM
 
+#define OE_WRITE_DEEPCOPY_OUT_PARAM(argname, argsize)                          \
+    do                                                                         \
+    {                                                                          \
+        void* p = (void*)(_deepcopy_out_buffer + _deepcopy_out_buffer_offset); \
+        OE_ADD_SIZE(_deepcopy_out_buffer_offset, (size_t)(argsize));           \
+        memcpy(p, argname, (size_t)(argsize));                                 \
+    } while (0)
+
+#define OE_SET_DEEPCOPY_OUT_PARAM(argname, argsize, argtype)               \
+    do                                                                     \
+    {                                                                      \
+        argname = (argtype)oe_malloc(argsize);                             \
```

Comment:
Does this handle case where a deep-copy member ptr is NULL?

Reply:
Updated the macro to be consistent with `OE_WRITE_IN_PARAM`